### PR TITLE
Interactive mode

### PR DESCRIPTION
**Pull request by sava-cska**
Thank you for contributing to KLEE.  We are looking forward to reviewing your PR.  However, given the small number of active reviewers and our limited time, it might take a while to do so.  We aim to get back to each PR within one month, and often do so within one week. 

To help expedite the review please ensure the following, by adding an "x" for each completed item:

- [ ] The PR addresses a single issue.  In other words, if some parts of a PR could form another independent PR, you should break this PR into multiple smaller PRs.
- [ ] There are no unnecessary commits. For instance, commits that fix issues with a previous commit in this PR are unnecessary and should be removed (you can find [documentation on squashing commits here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes)).
- [ ] Larger PRs are divided into a logical sequence of commits.
- [ ] Each commit has a meaningful message documenting what it does.
- [ ] The code is commented.  In particular, newly added classes and functions should be documented.
- [ ] The patch is formatted via  [clang-format](https://clang.llvm.org/docs/ClangFormat.html) (see also [git-clang-format](https://raw.githubusercontent.com/llvm/llvm-project/master/clang/tools/clang-format/git-clang-format) for Git integration).  Please only format the patch itself and code surrounding the patch, not entire files.  Divergences from clang-formatting are only rarely accepted, and only if they clearly improve code readability.
- [ ] Add test cases exercising the code you added or modified.  We expect [system and/or unit test cases](https://klee.github.io/docs/developers-guide/#regression-testing-framework) for all non-trivial changes.  After you submit your PR, you will be able to see a [Codecov report](https://docs.codecov.io/docs/pull-request-comments) telling you which parts of your patch are not covered by the regression test suite.  You will also be able to see if the Github Actions CI and Cirrus CI tests have passed.  If they don't, you should examine the failures and address them before the PR can be reviewed. 
- [ ] Spellcheck all messages added to the codebase, all comments, as well as commit messages.

### CODE REVIEW

File: lib/Runner/run_klee.cpp [hunk 1]
```diff
@@ -1613,189 +1864,78 @@ int run_klee(int argc, char **argv, char **envp) {
...
-      for (std::vector<std::string>::iterator it2 = kTestFiles.begin(),
-                                              ie = kTestFiles.end();
-           it2 != ie; ++it2) {
-        KTest *out = kTest_fromFile(it2->c_str());
-        if (!out) {
-          klee_error("unable to open: %s\n", (*it2).c_str());
+      entry_array.push_back(entrypoint);
+
+      if (entry_array.size() == PROCESS) {
+        for (std::string &entry : entry_array) {
```
Comment (ladisgin) [inline lib/Runner/run_klee.cpp:1892]: add const qualifier

File: lib/Runner/run_klee.cpp [hunk 2]
```diff
@@ -1613,189 +1864,78 @@ int run_klee(int argc, char **argv, char **envp) {
...
+          } else {
+            continue;
+          }
         }
-        seeds.push_back(out);
-      }
-      if (kTestFiles.empty()) {
-        klee_error("seeds directory is empty: %s\n", (*it).c_str());
+        entry_array.clear();
+        while (wait(NULL) > 0)
```
Comment (ladisgin) [inline lib/Runner/run_klee.cpp:1909]: Kill child processes, then kill KLEE

Comment (ladisgin) [inline lib/Runner/run_klee.cpp:1909]: Substitute current handle of function

File: lib/Runner/run_klee.cpp [hunk 3]
```diff
@@ -1613,189 +1864,78 @@ int run_klee(int argc, char **argv, char **envp) {
...
 
-    if (RunInDir != "") {
-      int res = chdir(RunInDir.c_str());
-      if (res < 0) {
-        klee_error("Unable to change directory to: %s - %s", RunInDir.c_str(),
-                   sys::StrError(errno).c_str());
-      }
+    if (ProcessNumber < 1 || ProcessNumber > MAX_PROCESS_NUMBER) {
+      klee_error("Incorrect number of process, your value is %d, but it must lie in [1, %d].",
+                 (int) ProcessNumber, MAX_PROCESS_NUMBER);
```
Comment (ladisgin) [inline lib/Runner/run_klee.cpp:1877]: change to getValue()

File: lib/Runner/run_klee.cpp [hunk 4]
```diff
@@ -1613,189 +1864,78 @@ int run_klee(int argc, char **argv, char **envp) {
...
-      switch (ExecutionMode) {
-      case ExecutionKind::Default:
-        interpreter->runFunctionAsMain(mainFn, out->numArgs, out->args, pEnvp);
-        break;
-      case ExecutionKind::Guided:
-        interpreter->runMainAsGuided(mainFn, out->numArgs, out->args, pEnvp);
-        break;
-      }
-      if (interrupted)
+    klee_message("Start %d processes.", (int) ProcessNumber);
```
Comment (ladisgin) [inline lib/Runner/run_klee.cpp:1879]: change cast to getValue()

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: lib/Runner/run_klee.cpp [hunk 5]
```diff
@@ -1613,189 +1864,78 @@ int run_klee(int argc, char **argv, char **envp) {
...
-      if (out) {
-        kTests.push_back(out);
-      } else {
-        klee_warning("unable to open: %s\n", (*it).c_str());
-      }
+  if (InteractiveMode) {
+    klee_message("KLEE finish preprocessing.");
+    std::ifstream entrypoints(EntryPointsFile);
+    if (!entrypoints.good()) {
+      klee_error("%s", ("Opening " + EntryPointsFile + " failed.").c_str());
```
[the remark this answers is not in the document]
Comment (sava-cska) [reply lib/Runner/run_klee.cpp:1872]: Fix

File: lib/Runner/run_klee.cpp (hunk 1 above)
Comment (sava-cska) [reply lib/Runner/run_klee.cpp:1892]: Ok, boomer

File: lib/Runner/run_klee.cpp (hunk 3 above)
Comment (sava-cska) [reply lib/Runner/run_klee.cpp:1877]: Ok

File: lib/Runner/run_klee.cpp (hunk 4 above)
Comment (sava-cska) [reply lib/Runner/run_klee.cpp:1879]: Ok

File: lib/Runner/run_klee.cpp (hunk 2 above)
Comment (sava-cska) [reply lib/Runner/run_klee.cpp:1909]: Change

Comment (sava-cska) [reply lib/Runner/run_klee.cpp]: Ok

**Review by ladisgin — APPROVED** [review]

**Pull request merged**